packages/test/functional-tests/src/test/ddsHandleEncoding.spec.ts

```diff
@@ -113,7 +107,7 @@ describe("DDS Handle Encoding", () => {
 		),
 		createTestCase(
 			new DirectoryFactory(),
-			(dds: IDirectory) => {
+			(dds: ISharedDirectory) => {
```

Comment:
That does not compile:

> Argument of type '(dds: IDirectory) => void' is not assignable to parameter of type '(dds: IChannel) => void'.
>   Types of parameters 'dds' and 'dds' are incompatible.
>     Type 'IChannel' is missing the following properties from type 'IDirectory'
> ...

My understanding is that ISharedDirectory is an alternative to IDirectory that is a ISharedObject (thats how its declared anyway). As this code is creating shared objects using DirectoryFactory which produces ISharedDirectory instances, it needs to take a ISharedDirectory.

Perhaps something is wrong elsewhere if this isn't how thats supposed to function?

Reply:
More specifically, what should DirectoryFactory produce? ISharedDirectory or IDirectory? Currently its implemented using the SharedDirectory which implements ISharedDirectory and not IDirectory.

Reply:
Weird, I wonder how this worked before then.  Probably either is fine then.

DirectoryFactory produces an ISharedDirectory.  ISharedDirectory extends IDirectory, in order to merge it with ISharedObject.

Reply:
ISharedDirectory does not extend IDirectory. It extends `Omit<IDirectory, "on" | "once" | "off">`

Reply:
If ISharedDirectory did extend IDirectory, then I think that code would have compiled with IDirectory

Reply:
Same idea though - it's only omitted from IDirectory because it collides with the event API surface on ISharedObject.  Hard to say without the full compile error.